developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+import re
+import subprocess
+
+from developers_chamber.scripts import cli
+from developers_chamber.utils import print_bold, print_error, print_heading, print_success
+from git import Repo
+
+
+class QAError(Exception):
+    def __init__(self, msg, output):
+        super().__init__(msg)
+        self.output = output.strip()
+
+
+class QACheck:
+    """
+    Base class for a quality assurance check.
+
+    Arguments:
+        name: Name of the check.
+    """
+    name = None
+
+    def _get_repo(self):
+        """
+        Returns the repo object.
+        """
+        return Repo('.')
+
+    def _get_default_branch(self):
+        """
+        Returns default branch of the repo.
+        """
+        return self._run_command("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'")
+
+    def _get_diffs(self):
+        """
+        Returns Git diffs against default branch.
+        """
+        repo = self._get_repo()
+        target_commit = repo.heads[self._get_default_branch()].commit
+        return target_commit.diff('HEAD', create_patch=True)
+
+    def _get_unstaged(self):
+        """
+        Returns unstaged files in the repo.
+        """
+        return self._get_repo().index.diff(None, create_patch=True)
+
+    def _run_command(self, command):
+        """
+        Runs shell command and returns its output.
+        """
+        try:
+            return subprocess.check_output(command, shell=True).decode().strip()
+        except subprocess.CalledProcessError as ex:
+            raise QAError(str(ex), ex.output.decode())
+
+    def _run_check(self):
+        """
+        This function should implement the actual check and should raise `QAError` if failed.
+        """
+        raise NotImplementedError()
+
+    def _cleanup(self):
+        """
+        Cleans up the repo to be fresh for another check.
+        """
+        self._get_repo().git.stash('save')
```

Comment:
I was thinking to use `git clean` but in order to be less destructive in case of accident, I am stashing the files.